job/job.go

```diff
@@ -0,0 +1,202 @@
+package job
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+
+	"github.com/gopheryan/jobby/streamer"
+)
+
+// Current process state
+type State string
+
+const (
+	// The process is currently running
+	JobStatusRunning State = "RUNNING"
+	// Not necessarily a 'success' (see exit code for that)
+	// This state signals that the process is no longer running
+	JobstatusComplete State = "COMPLETE"
+	// Different from 'COMPLETE' in that this state
+	// means that the user deliberately stopped the job
+	JobStatusStopped State = "STOPPED"
+)
+
+func newState(processExited, userKilled bool, e *exec.ExitError) State {
+	if processExited {
+		// ExitCode() returns -1 if the process hasn't exited or was terminated by a signal
+		// Well we know the process exited
+		if e != nil && e.ExitCode() == -1 {
+			status, ok := e.ProcessState.Sys().(syscall.WaitStatus)
+			// We know the process has exited, and the 'userKilled' flag *may*
+			// be set, but there's a chance that the process exited just as Stop()
+			// was called. Validate that this process was killed by a kill signal
+			if ok && status.Signal() == syscall.SIGKILL && userKilled {
+				// The process has exited, it was killed by a kill signal,
+				// and the job has a 'userKilled' flag indicating that the caller
+				// asked us to stop a running job.
+				return JobStatusStopped
+			}
+		}
+		return JobstatusComplete
+	}
+	return JobStatusRunning
+}
```

Comment:
Can this be simplified down to the following?
```suggestion
	if !processExited {
		return JobStatusRunning
	}
	
	if userKilled {
		return JobStatusStopped
	}
	
	return JobstatusComplete
}
```

Reply:
That would probably be fine. I had assumed (maybe incorrectly) that there was a possibility that the process could still exit gracefully after the call to `process.Kill`. And for that reason, I think there might be value in doing our due diligence to inspect the process state and determine if it was indeed stopped via a kill signal or had a graceful exit. 

It's a very tight window, and likely won't matter much to the user. I'm open to simplifying to this suggestion.

Reply:
Implemented